src/apps/concepts/components/ConceptForm.tsx

```diff
@@ -119,11 +119,30 @@ const initialValues: Concept = {
 const buildInitialValues = (
   conceptClass: string = initialValues.concept_class,
   names: ConceptName[]
-) => ({
-  ...initialValues,
-  concept_class: conceptClass,
-  names
-});
+) => {
+  // sort the names
+  names.sort((n1: ConceptName, n2: ConceptName) => {
+    if (n1.locale !== n2.locale) {
+      return n1.locale < n2.locale ? -1 : 1;
+    }
+    
+    if (n1.locale_preferred !== n2.locale_preferred) {
+      return n2.locale_preferred ? -1 : 1;
+    }
+
```

Comment:
Hopefully the last suggestion:

```suggestion

    if (n1.name_type !== n2.name_type) {
      if (n1.name_type === "FULLY_SPECIFIED") {
        return -1;
      } else if (n2.name_type === "FULLY_SPECIFIED") {
        return 1;
      }
    }

```